snuba/migrations/snuba_migrations/sessions/0003_sessions_matview.py

```diff
@@ -0,0 +1,139 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import (
+    UUID,
+    AggregateFunction,
+    Column,
+    DateTime,
+    String,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.processor import MAX_UINT32, NIL_UUID
+from .matview import create_matview_v1
+
+
+aggregate_columns: Sequence[Column[Modifiers]] = [
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("started", DateTime()),
+    Column("release", String(Modifiers(low_cardinality=True))),
+    Column("environment", String(Modifiers(low_cardinality=True))),
+    Column("user_agent", String(Modifiers(low_cardinality=True))),
+    Column("os", String(Modifiers(low_cardinality=True))),
+    Column("runtime", String(Modifiers(low_cardinality=True))),
+    # durations
+    Column(
+        "duration_quantiles",
+        AggregateFunction("quantilesIf(0.5, 0.9)", [UInt(32), UInt(8)]),
+    ),
+    Column("duration_avg", AggregateFunction("avgIf", [UInt(32), UInt(8)])),
+    # sessions:
+    Column("sessions", AggregateFunction("countIf", [UUID(), UInt(8)])),
+    Column("sessions_preaggr", AggregateFunction("sumIf", [UInt(32), UInt(8)])),
+    Column("sessions_crashed", AggregateFunction("countIf", [UUID(), UInt(8)])),
+    Column("sessions_crashed_preaggr", AggregateFunction("sumIf", [UInt(32), UInt(8)])),
+    Column("sessions_abnormal", AggregateFunction("countIf", [UUID(), UInt(8)])),
+    Column(
+        "sessions_abnormal_preaggr", AggregateFunction("sumIf", [UInt(32), UInt(8)])
+    ),
+    Column("sessions_errored", AggregateFunction("uniqIf", [UUID(), UInt(8)])),
+    Column("sessions_errored_preaggr", AggregateFunction("sumIf", [UInt(32), UInt(8)])),
+    # users:
+    Column("users", AggregateFunction("uniqIf", [UUID(), UInt(8)])),
+    Column("users_crashed", AggregateFunction("uniqIf", [UUID(), UInt(8)])),
+    Column("users_abnormal", AggregateFunction("uniqIf", [UUID(), UInt(8)])),
+    Column("users_errored", AggregateFunction("uniqIf", [UUID(), UInt(8)])),
+]
+
+
+query = f"""
+SELECT
+    org_id,
+    project_id,
+    toStartOfHour(started) as started,
+    release,
+    environment,
+    user_agent,
+    os,
+    runtime,
+
+    -- we would have liked to changed the quantiles here to `0.5, 0.75, 0.95, 0.99`:
+    quantilesIfState(0.5, 0.9)(
+        duration,
+        duration <> {MAX_UINT32} AND status == 1
+    ) as duration_quantiles,
+
+    -- this is new:
+    avgIfState(duration, duration <> {MAX_UINT32} AND status == 1) as duration_avg,
+
+    -- use the old `count` columns for "legacy" rows with `quantity == 0`:
+    countIfState(session_id, seq == 0 AND quantity == 0) as sessions,
+    countIfState(session_id, status == 2 AND quantity == 0) as sessions_crashed,
+    countIfState(session_id, status == 3 AND quantity == 0) as sessions_abnormal,
```

Comment:
As mentioned in the first PR in this stack. I think we can just remove these and avoit the artificial condition on `quantity == 0`. Clickhouse will not fill these columns and they will be treated as empty.